File: 2022/1/puzzle.py
"""provides path joining functions annd module joining"""
import os
import sys

INPUT_FILE="input"

def main():
  """Main Function called on Startup"""
  input_text= open(os.path.join(sys.path[0], INPUT_FILE),'r', encoding="UTF-8")
  lines = input_text.readlines()

  max_calories = [0,0,0]
  current_calories = 0
  for line in lines:
    if line.strip() == "":
      if max_calories[0] < current_calories:
        max_calories[0] = current_calories
        max_calories.sort()
      current_calories = 0
    else:
      current_calories += int(line.strip(), 10)
  if max_calories[0] < current_calories:
    max_calories[0] = current_calories
  print(f"{sum(max_calories)}")

if __name__ == "__main__":
  main()

File: 2022/1/puzzle.cs
using System;

namespace adventofcode {
  public class Day01 {
    private const string INPUTFILE = "input";
    public static void Run() {
      string[] input = File.ReadAllLines(Program.GetInputPath(INPUTFILE, "Day01"));
      int[] maxCalories = new int[]{0,0,0};
      int currentCalories = 0;
      foreach(string rawLine in input) {
        string line = rawLine.Trim();
        if (string.IsNullOrEmpty(line)) {
          if (maxCalories[0] < currentCalories) {
            maxCalories[0] = currentCalories;
            maxCalories = maxCalories.Order().ToArray();
          }
          currentCalories = 0;
        } else {
          currentCalories += int.Parse(line);
        }
      }
      if (maxCalories[0] < currentCalories) {
        maxCalories[0] = currentCalories;
        maxCalories = maxCalories.Order().ToArray();
      }
      Console.WriteLine($"{maxCalories.Sum()}");
    }
  }
}